static/js/notifications.js

```diff
@@ -217,14 +217,19 @@ exports.window_has_focus = function () {
     return window_has_focus;
 };
 
-function in_browser_notify(message, title, content) {
+function in_browser_notify(message, title, content, raw_operators, opts) {
     var notification_html = $(templates.render('notification', {gravatar_url: ui.small_avatar_url(message),
                                                                 title: title,
                                                                 content: content}));
     $('.top-right').notify({
         message: {html: notification_html},
         fadeOut: {enabled: true, delay: 4000}
     }).show();
+    $('.top-right').on('click', function () {
+        ui.change_tab_to('#home');
+        narrow.activate(raw_operators, opts);
+    });
+    setTimeout(function () {$('.top-right').unbind("click");}, 5000);
```

Comment:
For how many of these a user might have, it depends :)

So, how I would handle this, is store the stream+topic on the notification jquery object in a data field (grep for `data-stream-id` to see what I mean), and have a single click handler that's like a `$(body).on("click", ...)` style (similar to what's in `static/js/click_handlers.js`) for when someone clicks on anything matches a class the notifications have, which just reads the data field from `$(this)` (aka what they clicked on) to figure out what to narrow to.  Does that make sense?

With that approach, there's just one click handler, and it'll just trigger the notification once, and we don't need the setTimeout to remove the click handler, since it's just a single global handler for clicking something with that class.